src/Roles/Roles_Provider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Tribe\Libs\Roles;
+
+use Pimple\Container;
+use Tribe\Libs\Container\Service_Provider;
+use Tribe\Libs\Roles\Role\Administrator;
+use Tribe\Libs\Roles\Role\Author;
+use Tribe\Libs\Roles\Role\Editor;
+use Tribe\Libs\Roles\Role\Subscriber;
+
+class Roles_Provider extends Service_Provider {
+	const ROLES = 'users.roles';
+
+	public function register( Container $container ) {
+		$container[ self::ROLES ] = function () {
+			return [
+				new Author,
+				new Administrator,
+				new Editor,
+				new Subscriber,
+			];
+		};
+
+		$this->update_roles( $container );
+		$this->hide_topbar( $container );
+		$this->redirect_after_login( $container );
+		$this->restrict_admin_dashboard( $container );
+	}
+
+	private function update_roles( Container $container ) {
+		$container[ Roles_Schema::class ] = function () use ( $container ) {
+			return new Roles_Schema( $container[ self::ROLES ] );
+		};
+
+		add_action( 'admin_init', function () use ( $container ) {
+			if ( $container[ Roles_Schema::class ]->update_required() ) {
+				$container[ Roles_Schema::class ]->do_updates();
+			}
+		}, 0, 0 );
+	}
+
+	private function hide_topbar( Container $container ) {
+		add_filter( 'show_admin_bar', function (): bool {
+			return current_user_can( Capabilities::VIEW_ADMIN_DASHBOARD );
+		} );
+	}
+
+	private function redirect_after_login( Container $container ) {
+		$container[ Login_Redirector::class ] = function () use ( $container ) {
+			return new Login_Redirector( $container[ self::ROLES ] );
+		};
+
+		add_filter( 'login_redirect', function ( $redirect_to, $requested_redirect_to, $user ) use ( $container ) {
+			return $container[ Login_Redirector::class ]->redirect_after_login( $redirect_to, $requested_redirect_to,
+				$user );
+		}, 10, 3 );
+	}
+
+	private function restrict_admin_dashboard( Container $container ) {
+		$container[ Dashboard_Restictor::class ] = function () {
+			return new Dashboard_Restictor;
+		};
+
+		add_action( 'admin_init', function () use ( $container ) {
+			$container[ Dashboard_Restictor::class ]->check_user();
+		}, 0, 0 );
```

Comment:
```suggestion
		}, 1, 0 );
```

Reply:
Apply a fix to this PR. It should `check_user` should only run after `Roles_Schema::do_updates` has had a chance to fire. More info: https://tribe.slack.com/archives/C51FR5XL4/p1579965450010200?thread_ts=1579726625.001600&cid=C51FR5XL4